### ISSUE TITLE
Implement SGD optimizer with gradient clipping (Yang's optimizer)

### ISSUE THREAD

**Issue by mwinton** [mwinton/w266-final-project#70]
keras.optimizers.SGD(lr=?, momentum=?,  decay=?, nesterov=False, clipnorm=?)

From Yang's code:

options['lr'] = numpy.float32(0.1)
options['momentum'] = numpy.float32(0.9)
options['decay_rate'] = numpy.float32(0.999)
options['grad_clip'] = numpy.float32(0.1)

Note they do *not* use the separate `w_emb_lr` param.
